packages/java/parser-jvm-plugin-transfertypes/src/test/java/com/vaadin/hilla/parser/plugins/transfertypes/signals/SignalEndpoint.java

```diff
@@ -0,0 +1,25 @@
+package com.vaadin.hilla.parser.plugins.transfertypes.signals;
+
+import com.vaadin.hilla.signals.ListSignal;
+import com.vaadin.hilla.signals.NumberSignal;
+import com.vaadin.hilla.signals.Signal;
+import com.vaadin.hilla.signals.ValueSignal;
+
+@Endpoint
+public class SignalEndpoint {
+    public Signal<String> getStringSignal() {
+        return null;
+    }
```

Comment:
I think we should actually prevent this case from happening, though, it is possible in java.

Reply:
What do you mean? It's just a test file, isn't it?

Reply:
Hmm, that's a valid point but I believe it is out of scope for this PR since it is more about generics in general (pun intended), and signals are only an example for them.